nx/exercises/exercises-1-20.livemd

```diff
@@ -0,0 +1,174 @@
+# Exercises: 1-20
+
+```elixir
+Mix.install([{:nx, "~> 0.6"}])
+```
+
+## Introduction
+
+Inspired by the Python notebook _100 Numpy Exercises_:
+
+https://www.kaggle.com/code/utsav15/100-numpy-exercises/notebook
+
+## 1-10
+
+**1. Install Nx in a Livebook (★☆☆)**
+
+```elixir
+Mix.install([{:nx, "~> 0.6"}])
+```
+
+**2. Create a 1-D tensor of 10 zeros (★☆☆)**
+
+```elixir
+Nx.broadcast(0, {10})
+```
+
+**3. How to find the number of elements in a tensor (★☆☆)**
+
+```elixir
+Nx.tensor([[1, 2, 3], [4, 5, 6]])
+|> Nx.size()
+```
+
+**4. How to find the memory size of any tensor (★☆☆)**
+
+```elixir
+Nx.tensor([[1, 2, 3], [4, 5, 6]])
+|> Nx.byte_size()
+```
+
+**5. How to get the documentation of the `add/2` function from the command line? (★☆☆)**
+
+```elixir
+# I don't know if this is possible.
+# iex> h Nx.add
+```
+
+**6. Create a tensor of zeros of size 10 but the fifth value which is 1 (★☆☆)**
+
+```elixir
+zeros = Nx.broadcast(0, {10})
+index = Nx.tensor([4])
+Nx.indexed_put(zeros, index, 1)
+```
+
+**7. Create a tensor with values ranging from 10 to 49 (★☆☆)**
+
+```elixir
+Nx.linspace(10, 49, n: 39, type: :s8)
+```
+
+**8. Reverse a tensor (first element becomes last) (★☆☆)**
+
+```elixir
+Nx.iota({10})
+|> Nx.reverse()
+```
+
+**9. Create a 3x3 tensor with values ranging from 0 to 8 (★☆☆)**
+
+```elixir
+Nx.iota({3, 3})
+```
+
+**10. Find indices of non-zero elements from [1,2,0,0,4,0] (★☆☆)**
+
+```elixir
+Nx.tensor([1, 2, 0, 0, 4, 0])
+|> Nx.not_equal(0)
+```
```

Comment:
Hey 👋 
I was just checking the PR (great work!)

I think this solution does not respect the exercise description because it returns this tensor `[1, 1, 0, 0, 1, 0]` instead of the indices of non-zero elements `[0, 1, 4]`.

This should work, although there might be better way to achieve that 🤔 

```suggestion
```elixir
t = Nx.tensor([1, 2, 0, 0, 4, 0])
to_keep = Nx.not_equal(t, 0)
indices = Nx.iota(Nx.shape(t))

Nx.select(to_keep, indices, -1)
|> Nx.to_list()
|> Enum.reject(& &1 == -1)
```
```

Cheers ✌️